### PR TITLE
Changes to error handling on adding attachments to trx

### PR DESCRIPTION
**Pull request by prawalagarwal**
Today if there is any exception while adding test result attachments to the trx, then the stack is not logged and only the message is logged. This is creating a problem in debugging the problems where an exception is thrown while attaching an attachment with the trx.

This PR will change the above issue. We will now log the exception stack which will help in debugging the issue.
[https://mseng.visualstudio.com/VSOnline/_workitems/edit/1197456](BUG 1197456)

### CODE REVIEW

**Review by smadala — CHANGES_REQUESTED** [review]

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/Converter.cs [hunk 1]
```diff
@@ -387,13 +387,12 @@ private static void UpdateTestResultAttachments(ObjectModel.TestResult rockStead
...
                         attachmentSet.DisplayName, 
                         e.GetType().ToString(), 
-                        e.Message);
+                        e);
 
                     StringBuilder stdErr = new StringBuilder(testResult.StdErr);
                     stdErr.AppendLine(errorMsg);
 
                     testResult.StdErr = stdErr.ToString();
-                    testResult.Outcome = TrxObjectModel.TestOutcome.Error;
```
[the remark this answers is not in the document]
Comment (prawalagarwal) [reply src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/Converter.cs]: Added

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/Converter.cs [hunk 2]
```diff
@@ -387,13 +387,12 @@ private static void UpdateTestResultAttachments(ObjectModel.TestResult rockStead
                         TrxLoggerResources.FailureToAttach,
                         attachmentSet.DisplayName, 
                         e.GetType().ToString(), 
-                        e.Message);
+                        e);
```
[the remark this answers is not in the document]
Comment (prawalagarwal) [reply src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/Converter.cs]: Done

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/Converter.cs [hunk 3]
```diff
@@ -387,13 +393,19 @@ private static void UpdateTestResultAttachments(ObjectModel.TestResult rockStead
...
+                    {
+                        ObjectModel.EqtTrace.Error("Converter: UpdateTestResultAttachments: " + errorMsg);
+                    }
 
                     StringBuilder stdErr = new StringBuilder(testResult.StdErr);
                     stdErr.AppendLine(errorMsg);
 
                     testResult.StdErr = stdErr.ToString();
                     testResult.Outcome = TrxObjectModel.TestOutcome.Error;
+                    throw e;
```
Comment (smadala) [inline src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/Converter.cs]: Add rethrow `throw;`.

**Review by smadala — APPROVED** [review]
- Update the description add we are not changing to Error.

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/Converter.cs (hunk 3 above)
Comment (prawalagarwal) [reply src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/Converter.cs]: Removing throw. Keeping the behavior same as before

**Pull request merged**